3rdparty/kleidicv/CMakeLists.txt

```diff
@@ -2,6 +2,10 @@ project(kleidicv_hal)
 
 if(HAVE_KLEIDICV)
   option(KLEIDICV_ENABLE_SME2 "" OFF) # not compatible with some CLang versions in NDK
-  ocv_warnings_disable(CMAKE_CXX_FLAGS -Wno-unused-function)
   include("${KLEIDICV_SOURCE_PATH}/adapters/opencv/CMakeLists.txt")
+  # HACK to suppress adapters/opencv/kleidicv_hal.cpp:343:12: warning: unused function 'from_opencv' [-Wunused-function]
+  target_compile_options( kleidicv_hal PRIVATE
+      $<TARGET_PROPERTY:kleidicv,COMPILE_OPTIONS>
+      "-Wno-old-style-cast" "-Wno-unused-function"
```

Comment:
Need to include through `-isystem` to avoid warnings from these external headers.

https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html
https://gcc.gnu.org/onlinedocs/cpp/System-Headers.html

Example:
https://github.com/opencv/opencv/blob/4.10.0/3rdparty/protobuf/CMakeLists.txt#L154

Reply:
Why does this widely used approach not work?

https://github.com/opencv/opencv/blob/4.10.0/3rdparty/libjpeg-turbo/CMakeLists.txt#L11

Reply:
I tried it. Most probably the issue is that the ocv_disable_warning works well inside the cmake "project" scope, but does not outside.